main/util.py

```diff
@@ -214,16 +216,38 @@ def create_name_from_email(email):
   return re.sub(r'_+|-+|\.+|\++', ' ', email.split('@')[0]).title()
 
 
-def password_hash(user_db, password):
+def password_hash(password, salt):
   m = hashlib.sha256()
-  m.update(user_db.key.urlsafe())
-  m.update(user_db.created.isoformat())
-  m.update(m.hexdigest())
   m.update(password.encode('utf-8'))
-  m.update(config.CONFIG_DB.salt)
+  m.update(salt)
   return m.hexdigest()
 
 
+def password_salt(n_bytes=32):
+  return binascii.hexlify(os.urandom(n_bytes))
+
+
+# Taken from Django Source Code
+# Helps prevent timing attacks
+# See http://carlos.bueno.org/2011/10/timing.html
+def constant_time_compare(val1, val2):
```

Comment:
Actually on a second thought.. yes you are right if you trying to crack the actual password if we were storing the plain text password.. but but but we are not comparing the naked passwords.. and by adding or removing random characters we are not exposing anything.. the times will be completely random and in practice the comparison will be ending very very fast, after comparing maximum the first couple of characters.. :) and you will achieve nothing.. 

(you can try it yourself by comparing only the the sha1 values by adding the one character at the time to get to the actual password..)